### PR TITLE
Cache fstab fetching

### PR DESCRIPTION
**Pull request by trieloff**
This PR adds a simple memoization function based on a LRU cache that remembers the FSTabs for the last 1000 repo, owner, ref pairs.

### CODE REVIEW

File: src/cache.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+const QuickLRU = require('quick-lru');
+
+let lru = new QuickLRU({ maxSize: 1000 });
+
+/**
+ * Returns a memoized version of the function `fn`.
+ * @param {function} fn a function to memoize
+ * @param {function} hash a hash function to build a cache key.
```
Comment (tripodsan) [inline src/cache.js:20]: better to use a dedicated. `options` argument.
- easier to document
- easier to create TS decls
- easier to understand.
- easier for IDEs to typecheck

File: src/cache.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      // if it's cached, just return it
+      const result = lru.get(key);
+      if (result.ok) {
+        return result.ok;
+      }
+      throw result.err;
+    }
+    try {
+      // invoke the function
+      const ok = await fn(...args);
```
Comment (tripodsan) [inline src/cache.js:50]: `ok` ?? :-)

File: src/github.js [hunk 3]
```diff
@@ -31,7 +32,7 @@ function computeGithubURI(root, owner, repo, ref, path) {
   return URL.format(rootURI);
 }
 
-async function fetchFSTab(root, owner, repo, ref, log, options) {
+async function fetchFSTabImpl(root, owner, repo, ref, log, options) {
```
Comment (tripodsan) [inline src/github.js:35]: still a very long list of arguments. please convert into options

File: src/cache.js (hunk 1 above)
Comment (trieloff) [reply src/cache.js:20]: The `cache` function is already using an `options` argument as the second argument. Nesting the function in the options argument, too doesn't make much sense.

**Pull request merged**